src/commands/oas2tson.js

```diff
@@ -0,0 +1,149 @@
+  }
+}
+
+const processSchema = (schema, schemasPath, definitionKeyword, isArray) => {
+  Object.entries(schema).forEach(([key, value]) => {
+    // for elements in an array the name would be its index if we were
+    // to just use its key, so go into the parsed schema and get the
+    // actual name so the files are more easily identifiable
+    const name = isArray ? value.name : key
+    const filename = _trimStart(filenamify(name, { replacement: '-' }), '-')
+
+    adaptSchema(value, name, filename)
+
+    let schemaAsString = JSON.stringify(value, null, 2)
+    // N.B. - this obviously only supports refs where the string contains 'components/schemas'
+    // if we want to support refs in places other than this, we'll need to revisit this
+    // approach to be more flexible
+    const refs = schemaAsString.match(COMPONENT_REF_REGEXP)
+    refs?.forEach(ref => {
+      let refName = ref.split('/').slice(-1)
+      schemaAsString = schemaAsString.replace(ref, `${refName}.json`)
+    })
+
+    const destinationDir = path.join(schemasPath, 'tempjson')
+    const destinationPath = path.join(destinationDir, `${filename}.json`)
+
+    outputSchemasMetaData.push({ dir: destinationDir, path: destinationPath })
+
+    fs.ensureDirSync(destinationDir)
+    fs.writeFileSync(destinationPath, schemaAsString)
+  })
+}
+
+const processJSON = async schemasPath => {
+  for (const currentSchema of outputSchemasMetaData) {
+    const dereferencedSchema = await $RefParser.dereference(currentSchema.path)
+    const fileName = path.parse(currentSchema.path).name
+    const tsSchema = `export const ${fileName} = ${JSON.stringify(
+      dereferencedSchema
+    )} as const`
+    const formattedSchema = await prettier.format(tsSchema, {
+      parser: 'typescript'
+    })
+    fs.ensureDirSync(schemasPath)
+    fs.writeFileSync(path.join(schemasPath, `${fileName}.ts`), formattedSchema)
+  }
+  fs.removeSync(path.join(schemasPath, 'tempjson'))
+}
+
+export const runCommand = async (
+  openApiPath,
+  schemasPath,
+  propertiesToExport,
+  logger = pino()
+) => {
+  fs.removeSync(schemasPath)
+  fs.ensureDirSync(schemasPath)
+
+  let openAPIContent
+
+  try {
+    openAPIContent = fs.readFileSync(openApiPath, 'utf8')
+  } catch (e) {
+    logger.error('❌ Could not find the OpenAPI file')
+    exit(1)
+  }
+
+  const parsedOpenAPIContent = YAML.parse(openAPIContent)
+
+  const definitionKeywords = [
+    ...new Set([
+      ...(propertiesToExport?.split(',') || []),
+      'components.schemas'
+    ])
+  ]
+
+  try {
+    const generatedSchema = fromSchema(parsedOpenAPIContent, {
+      definitionKeywords
+    })
+
+    definitionKeywords.forEach(key => {
+      const schema = _get(generatedSchema, key)
+      const isArray = Array.isArray(_get(parsedOpenAPIContent, key))
+      processSchema(schema, schemasPath, key, isArray)
+    })
+    await processJSON(schemasPath)
+  } catch (error) {
+    logger.warn('Failed to convert non-object attribute, skipping')
```

Comment:
can you log a bit more?
like: what is the attribute that wasn't processed?

Reply:
You should invoke the `warn` logger function once

Reply:
Noted, changed the warning